[MIG][website_mass_mailing_name] Migration to v10

yajo opened the pull request:
- Relicense to LGPL.
- Fix all known issues.
- Enable tour only in demo instances.
- Fix Sass headers.
- Remove compiled css and maps.
- Update JS modules to new api.
- Update tour to new tours api.
- Update module structure to match latest template.

@Tecnativa

yajo committed:
[8.0][ADD][website_snippet_mass_mailing_partner] Snippet + name

Let you have the name field in the mass mailing subscription snippet.

Leave most logic to base module.

Make name field autofilled if user data is available.

Add tour test.

yajo committed:
Workaround test failure

Test does not work, probably due to the framework. Leaving out the last step and hope for the best for v9.

oca-transbot committed:
OCA Transbot updated translations from Transifex

yajo committed:
[FIX][website_mass_mailing_name] Show disabled name. (#83)

[FIX]website_mass_mailing_name: Show disabled name.

When the snippet appears as disabled, do not hide the name field.

yajo committed:
[MIG][website_mass_mailing_name] Migration to v10

- Relicense to LGPL.
- Fix all known issues.
- Enable tour only in demo instances.
- Fix Sass headers.
- Remove compiled css and maps.
- Update JS modules to new api.
- Update tour to new tours api.
- Update module structure to match latest template.

pedrobaeza commented:
Can't try on runbot and Travis is red (maybe not because this module), but it's difficult to get a correct interpretation. Let's try to fix the branch first.

yajo commented:
Seems the [lack of sass](https://travis-ci.org/OCA/social/jobs/250665214#L1000) again.

chienandalu commented:
It still fails. Runbot:
```"LoadError: cannot load such file -- bootstrap-sass
 
 This error occured while compiling the bundle 'web.assets_frontend' containing:
     - /web/static/lib/bootstrap/less/variables.less
     - /web/static/lib/bootstrap/less/mixins/vendor-prefixes.less
     - /web/static/lib/bootstrap/less/mixins/buttons.less
     - /web/static/src/less/variables.less
     - /web/static/src/less/utils.less
     - /web_editor/static/src/less/web_editor.variables.less
     - /web/static/src/less/import_bootstrap.less
     - /web_editor/static/src/less/web_editor.common.less
     - /website/static/src/less/website.ui.less
     - /website/static/src/less/website.ui.components.less
     - /website/static/src/less/website.wrapwrap.less
     - /website/static/src/less/website.snippets.less
     - /website/static/src/less/website.zoomodoo.less
     - /website_mass_mailing_name/static/src/css/website_mass_mailing_name.sass"```

yajo commented:
Ahm of course, this adds sass to travis, not to runbot. How do I do that?

pedrobaeza commented:
We are using travis2docker on runbot, but that kind of packages is not translated. Is there any way to install by "command line"?

yajo commented:
I'm getting [this](https://travis-ci.org/OCA/social/jobs/251900547#L220-L221):

```
$ gem install bootstrap-sass
Fetching: execjs-2.7.0.gem (100%)
Successfully installed execjs-2.7.0
Fetching: autoprefixer-rails-7.1.2.1.gem (100%)
ERROR:  Error installing bootstrap-sass:
	autoprefixer-rails requires Ruby version >= 2.0.
```

Maybe the Travis gurus @moylop260 or @lasley can help here...

lasley commented:
We need to make this easier. There's actually three required lines. I'll make the suggestion to sample travis, but here there are:
``` yaml
before_install:
 - rvm install ruby --latest
 - gem install bootstrap-sass
 - gem install compass --pre
```

moylop260 commented:
Offtopic: Could you review the beta js lint https://travis-ci.org/OCA/social/jobs/250665212#L527-L533?

lasley commented:
Interesting this one is new to me `[W7903(javascript-lint), ] Expected dot to be on same line as object. [Error/dot-location]`. That makes me sad, I always put the dot prefacing the method (on the lower line) when using multiple lines

moylop260 commented:
Do you recommend us disable it?

lasley commented:
We should probably leave enabled - IMO if it's in a Lint engine & I don't like it, I just don't know enough of the intricacies of the language to understand why it's a rule. 

Basically I've just been doing it wrong in ignorance for numerous years, so it's just going to be one of those things that feels clunky for a while & cause me to gripe 😆

moylop260 committed:
[REF] .travis.yml: Using shortcut to install phantomjs 2, ruby 2 and compass
Ruby 2 fix https://github.com/Compass/compass/issues/1513#issuecomment-32276852
Phantomjs 2 fix `odoo.tests.common: phantomjs: u"ReferenceError: Can't find variable: MutationObserver`

moylop260 commented on website_mass_mailing_name/tests/test_ui.py:
```diff
@@ -0,0 +1,15 @@
+        """Test snippet behavior."""
+        self.phantom_js(
+            "/",
+            "openerp.Tour.run('mass_mailing_partner', 'test')",
```
`odoo.__DEBUG__.services['web_tour.tour'].run('mass_mailing_partner', 'test')` & `odoo.__DEBUG__.services['web_tour.tour'].tours.mass_mailing_partner.ready` for v10.0?

yajo replied:
This is now included in the migration commit, thanks!

moylop260 commented:
@yajo I just updated the .travis.yml in the last commit 🤕 
@chienandalu Could you test it again from runbot, please?

I tested the newsletter from https://runbot.odoo-community.org/runbot/205/179

![testing](https://user-images.githubusercontent.com/6644187/28040137-6133c126-658a-11e7-8a9a-0a0930e3cd1d.png)

chienandalu reviewed (commented):
Everything seem to work fine. There's a strange behavior though:
- I inserted a block and chose the newsletter I wanted to point to.
- Entered from an anonymous session to the website and subscribed. I get the acknowledge and the email is subscribed.
- I refresh the page and the subscription form is now blocked:
![image](https://user-images.githubusercontent.com/5040182/28060011-39abf88e-6626-11e7-8036-8f5b87f2b3f6.png)
- This also happens if a logged in user subscribes with his system email address.
- If a logged in user subscribes any other email address the form doesn't block.

Seems also that some of your tests may be failing.

yajo commented:
This strange behaviour you see @chienandalu is Odoo's default thing. Just try in a bare runbot and you will see that it happens the same, with the difference being that there's no name field.

Also I fixed the tests, the tour definition was plain wrong, the tour call was wrong too, and besides it turns out you cannot switch the user in the middle of new v10 tours, I had to split it in logged tour and public tour, and run both sequentially.

I also rebased to get rid of #181, so in short it should go ✔️ now

pedrobaeza commented:
This is still red

yajo commented:
@moylop260 it seems compass, sass, or bootstrap-sass is not correctly installed. Could you check please?

moylop260 commented:
It seems similar to https://github.com/OCA/website/pull/361#issuecomment-316182210
I think that you should use the exactly version of these packages when worked

moylop260 commented:
@Yajo 
If locally works could you send us the packages versions, please?
 - `ruby --version`
 - `gem list`

The issues related comments are about versions:
 - https://github.com/Compass/compass/issues/1513#issuecomment-32276852

I just updated to `ruby 2.0` like as recommendation

moylop260 commented:
@Yajo 
Now we have the error 
```bash
2017-07-19 00:42:29,361 14623 WARNING openerp_test odoo.addons.base.ir.ir_qweb.assetsbundle: NoMethodError: undefined method `activate' for nil:NilClass
```

https://travis-ci.org/OCA/social/jobs/255080735#L1080

But the fixes now talk about the implementation of code
https://github.com/asciidoctor/asciidoctor-pdf/commit/b6a0df9e06a9b5640e6c68634cb8a745f46e558d

Could you review it, please?

lasley commented:
Shortcut for this added in https://github.com/OCA/maintainer-quality-tools/pull/465

yajo commented:
Let's see if it goes 💚 now

moylop260 commented:
The error `odoo.tests.common: phantomjs: u"ReferenceError: Can't find variable: MutationObserver`  is reproduced from runbot and It seems a phantomjs version error:

 - <img width="1278" alt="screen shot 2017-07-20 at 14 52 54" src="https://user-images.githubusercontent.com/6644187/28435575-2f96b0fc-6d59-11e7-8b4e-4ad155db905b.png">

@Yajo you faced all posible environment errors in a week 😆 

I will update your travis.yml to fix it

cc @gurneyalex FYI

moylop260 commented:
Ok, now is red because other matter...

Currently MQT don't use `--db-filter` parameter to run testing:
 - https://travis-ci.org/OCA/social/jobs/255836472#L1046

For phantomjs test we need just one database using the `--db-filter` parameter (because the UI change if 2 database are enabled to connect) so we can fix it from MQT:

 - <img width="1275" alt="screen shot 2017-07-20 at 18 18 52" src="https://user-images.githubusercontent.com/6644187/28442571-89d257c0-6d76-11e7-9d2c-19b09274a6cb.png">


In fact, the feature is working well!

 - [![IMAGE ALT TEXT HERE](https://img.youtube.com/vi/XrJs-BUoCYo/0.jpg)](https://www.youtube.com/watch?v=XrJs-BUoCYo)

moylop260 reviewed (approved):
Dummy commit using https://github.com/OCA/maintainer-quality-tools/pull/468 worked fine!

Travis is green 🎉 
 - <img width="973" alt="screen shot 2017-07-20 at 19 44 23" src="https://user-images.githubusercontent.com/6644187/28444311-dfc60c3e-6d81-11e7-9559-b5dbcf4cce2e.png">

(The LINT build error is unrelated https://github.com/OCA/maintainer-quality-tools/issues/467)
